pennylane/circuit_graph.py

```diff
@@ -469,9 +471,57 @@ def get_depth(self):
                 self._operation_graph = self._graph.subgraph(
                     list(self._graph.nodes().index(node) for node in self.operations)
                 )
-                self._depth = rx.dag_longest_path_length(self._operation_graph) + 1
+                self._extend_graph(self._operation_graph)
+                self._depth = (
+                    rx.dag_longest_path_length(
+                        self._operation_graph, weight_fn=lambda _, __, w: self._weight_func(w)
+                    )
+                    + 1
+                )
         return self._depth
 
+    @staticmethod
+    def _weight_func(weight):
+        """If weight is a number, use it!"""
+        if isinstance(weight, Number):
+            return weight
+        return 1
+
+    def _extend_graph(self, graph: rx.PyDiGraph) -> rx.PyDiGraph:
+        """Extend graph to account for custom depth operations"""
+        custom_depth_node_dict = {}
+        for op in self.operations:
+            if isinstance(op, ResourcesOperation) and (d := op.resources().depth) > 1:
```

Comment:
In the future, might there be any other objects that define custom resources? Any chance this will not catch that an `op` has `depth>1`?

Reply:
I see what you mean! I think if any op is to define custom resources, than we can safely expect the user to use `ResourcesOperation` class to do it. That is the standard pathway to do it.